### PR TITLE
feat: support casing window string operation

### PR DESCRIPTION
**Pull request by ovineq**
Introduces casing windows on the intersection:
image

Related data change: https://github.com/equinor/esv-intersection-data/pull/12

Issue: https://github.com/equinor/wellx-designer/issues/715

### CODE REVIEW

**Review by ooystein — CHANGES_REQUESTED** [review]
Getting an error in storybook so can not test PR:
image

**Comment by ooystein** [comment]
You should play around with git and learn how you could "easily" split different concerns into different branches/PRs :face_with_spiral_eyes:

It would make it much easier to review this PR if the IPoint changes (and maybe som other unrelated changes) was split out.

But I'm always happy to see developers leaving the codebase in a better shape than they found it in! Keep it up :grin:

**Comment by ovineq** [comment]
> Getting an error in storybook so can not test PR: image

I don't experience the same issues locally. Did you checkout the linked data changes from the description? 😅

**Comment by ovineq** [comment]
> You should play around with git and learn how you could "easily" split different concerns into different branches/PRs face_with_spiral_eyes
> 
> It would make it much easier to review this PR if the IPoint changes (and maybe som other unrelated changes) was split out.
> 
> But I'm always happy to see developers leaving the codebase in a better shape than they found it in! Keep it up grin

Good points. I'll be sure to split it out into its own PR next time.

File: package.json [hunk 1]
```diff
@@ -103,6 +103,7 @@
     "d3-scale": "^4.0.2",
     "d3-selection": "^3.0.0",
     "d3-shape": "^3.1.0",
-    "d3-zoom": "^3.0.0"
+    "d3-zoom": "^3.0.0",
+    "pixi-dashed-line": "^1.4.2"
```
[the remark this answers is not in the document]
Comment (ovineq) [reply package.json:107]: Good catch. Pixi is also a peer of `pixi-dashed-line`, so I'll add it there.

File: src/layers/CustomDisplayObjects/ComplexRope.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
-import { IPoint, Mesh, MeshMaterial, Renderer, Texture, WRAP_MODES } from 'pixi.js';
+import { Mesh, MeshMaterial, Point, Renderer, Texture, WRAP_MODES } from 'pixi.js';
 import { ComplexRopeGeometry } from './ComplexRopeGeometry';
 
 export type ComplexRopeSegment = {
-  points: IPoint[];
+  points: Point[];
```
[the remark this answers is not in the document]
Comment (ovineq) [reply src/layers/CustomDisplayObjects/ComplexRope.ts:5]: We have some `.clone()` calls which is not specified in the `IPoint` interface. That's the reasoning behind changing some of the `IPoint` to `Point`.
But it seems this particular change in the `ComplexRopeSegment` was unnecessary and is now reverted. 😄

**Review by ooystein — APPROVED** [review]
Excellent stuff

**Pull request merged**